### DISCUSSION TITLE
Use with local TeXLive installation

### DISCUSSION THREAD

**Discussion by fsimonis** [tectonic-typesetting/tectonic#1187] [Q&A]
Can I use tectonic solely with a local TeXLive installation and without any bundles?

Are there any downsides to doing this if reproducibility isn't an issue?

**Comment by mathlab08** [comment]
For Windows users
(1) Download https://data1.fullyjustified.net/tlextras-2022.0r0.tar, https://data1.fullyjustified.net/tlextras-2022.0r0.tar.index.gz and put it in the directory TeXLive
(2) Download nginx-1.26.0.zip, unzip it into the directory nginx-1.26.0
(3) Download Tectonic.exe, put it into the directory Tectonic, and create the subdirectory TectonicProject
(4) Create batch file cmd-input-latex-file.bat

```
@echo on
color 3e

setlocal
set TECTONIC_CACHE_DIR=%~dp0Tectonic\TectonicProject\Tectonic
set PATH=%~dp0\Tectonic;%SystemRoot%\system32;%PATH%
:: Get the absolute path to the current directory
set CURRENT_DIR=%cd%

:: Set the path to the Nginx configuration file
set NGINX_CONFIG_PATH=nginx.conf

:: Create or update the Nginx configuration file.
(
echo events {
echo   worker_connections 1024;
echo }
echo http {
echo     include       mime.types;
echo     default_type  application/octet-stream;
echo     sendfile        on;
echo     keepalive_timeout  65;
echo server {
echo     listen       8000;
echo     server_name  localhost 127.0.0.1 0.0.0.0;   
echo     location / {
echo         root   %CURRENT_DIR%\TeXLive;
echo         index  index.html index.htm;
echo     }
echo   }
echo }
) >%CURRENT_DIR%\nginx-1.26.0\conf\%NGINX_CONFIG_PATH%
cd %CURRENT_DIR%\nginx-1.26.0

:: Start Nginx
start nginx

::Enter the LaTeX file to be compiled.
echo ============
cd ...
echo Enter the name of the latex file to be compiled (including the path but not the extension)
set /p FileName=">"
tectonic.exe -w http://localhost:8000/tlextras-2022.0r0.tar -X compile %FileName%.tex
echo ============
pause
taskkill /F /IM nginx.exe
endlocal

```

All the above directories and batch files into the same home directory LocalTectonicTeXLive.
(5) Double-click cmd-input-latex-file.bat to start the Nginx server, and enter the name of the latex file to be compiled (including the path but not the extension) in the command line window, then enter to compile.
(6) After compiling, press any key to exit Nginx server.
